README.md

```diff
@@ -0,0 +1,211 @@
+
+Only Java version 8 is supported at the moment.
+
+### Install Bouncy Castle provider
+
+The Bouncy Castle library is required to be installed in the JRE in order to run the Java utility.
+
+See:
+https://github.com/wultra/powerauth-server/wiki/Installing-Bouncy-Castle
+
+### Generate a Signing Key Pair
+
+Before signing the SSL certificate, you will need to generate a new ECDSA key pair and store it in the `keypair.pem` file. You need to store this file securely. This key pair will be used to sign current certificate and all future certificates. If you already signed a certificate before, skip this step and use the previously generated key pair.
+
+The `keypair.pem` contains the private key. The public key can be printed as described in chapter Export public key. 
+
+The following command generates the key pair in PEM format. The key pair is protected by password of your choice. 
+
+```sh
+java -jar ssl-pinning-tool.jar keygen -o keypair.pem -p [password]
+```
+
+**Store the key pair and private key password safely! You will need it next time you replace SSL certificate to generate new signatures.**
+
+### Prepare Certificate Signature
+
+This command will retrieve the SSL certificate into PEM file:
+
+```sh
+openssl s_client -showcerts -connect my.domain.com:443 -servername my.domain.com < /dev/null | openssl x509 -outform PEM > cert.pem
+```
+
+Make sure to replace `my.domain.com` in both `-connect` and `-servername` options with your domain name.
+
+This command will generate the SSL certificate signature:
+
+```sh
+java -jar ssl-pinning-tool.jar sign -k keypair.pem -c cert.pem -o output.json -p [password]
+```
+
+The output file will contain SSL certificate signature:
+```json
+{
+  "name" : "my.domain.com",
+  "fingerprint" : "jymEKdgGPv1zSp61CYya5c2fR9fTLe8tKnWF6857iLA=",
+  "expires" : 1543322263000,
+  "signature" : "MEUCICOs9bb6TIEmRNHCekxn9URADLYuuZnk4aftpVDzdwmWAiEAlU2r9VDEnAWryxvbAsSJfIlCQjKfumdFbZeUKda166w="
+}
+``` 
+
+Alternatively, you can prepare certificate signature from information about the certificate. This approach is useful when the certificate is not deployed yet.
+
+You need following information:
+* Domain common name, e.g. `my.domain.com`
+* Certificate fingerprint in HEX format, for example: `8f298429d8063efd734a9eb5098c9ae5cd9f47d7d32def2d2a7585ebce7b88b0`
+* SSL certificate expiration time as Unix timestamp, e.g. `1543322263000`
+
+This command will generate the SSL certificate signature:
+
+```sh
+java -jar ssl-pinning-tool.jar sign -k keypair.pem -f 8f298429d8063efd734a9eb5098c9ae5cd9f47d7d32def2d2a7585ebce7b88b0 -t 1543322263000 -n my.domain.com -o output.json -p [password]
+```
+
+The output file will contain JSON with SSL certificate signature:
+```json
+{
+  "name" : "my.domain.com",
+  "fingerprint" : "jymEKdgGPv1zSp61CYya5c2fR9fTLe8tKnWF6857iLA=",
+  "expires" : 1543322263000,
+  "signature" : "MEUCICOs9bb6TIEmRNHCekxn9URADLYuuZnk4aftpVDzdwmWAiEAlU2r9VDEnAWryxvbAsSJfIlCQjKfumdFbZeUKda166w="
+}
+``` 
+
```

Comment:
We are missing information about final JSON with multiple signatures. This should probably be a reference to the last chapter?

Reply:
This information is at the very bottom of the documentation file, chapter "Prepare the JSON with Signature". It is shared between both Java tool and Open SSL scripts.

Reply:
Yes, I just thought we might mention the last chapter here, so that the single-signature JSON is not considered as the final JSON.